samples: matter: Align changes in all samples to sysbuild

ArekBalysNordic opened the pull request:
- Removed redundant `_release` pm_static files.
- Moved Matter diagnostic log snippets to theNRF common snippet directory.
- Simplified and reorganized kconfig.sysbuild files.
- Aligned Matter docs to sysbuild changes.
- Added several entries to the migration guide.

This PR repairs multiple bugs in Matter regarding sysbuild.

Depends on https://github.com/nrfconnect/sdk-nrf/pull/15340

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-15686/1/)
- Integration Platforms

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-15686/1/)
| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| test-fw-nrfconnect-chip | X | | |
| test-sdk-dfu | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-15686/1/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

wiba-nordic commented on applications/matter_weather_station/README.rst:
```diff
@@ -204,26 +204,25 @@ To build this application to work with the nRF7002 Wi-Fi expansion board:
 =======================
 
-To enable factory data support, you need to select the ``overlay-factory_data`` configuration overlay from the available application :ref:`build configuration overlays <matter_weather_station_app_build_configuration_overlays>`.
+To enable factory data support, you need to select the ``overlay-factory_data`` configuration overlay from the available application :ref:`build configuration overlays <matter_weather_station_app_build_configuration_overlays>`, set the :kconfig:option:`SB_CONFIG_MATTER_FACTORY_DATA` kconfig option to ``y``, and switch the file suffix to ``factory_data``.
```
```suggestion
To enable factory data support, you need to select the ``overlay-factory_data`` configuration overlay from the available application :ref:`build configuration overlays <matter_weather_station_app_build_configuration_overlays>`, set the :kconfig:option:`SB_CONFIG_MATTER_FACTORY_DATA` kconfig option to ``y``, and use the ``factory_data`` file suffix.
```
I think this is correct, since a suffix isn't required by default so switching might not be the case.

wiba-nordic commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -71,6 +71,19 @@ Peripheral samples
+  * With the inheritance of Zephyr's :ref:`zephyr:sysbuild` in the |NCS|, some changes are provided to the Matter samples and applications:
+
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD` kconfig option is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_FACTORY_DATA` instead to enable or disable creating the factory data set during building a Matter sample.
+      To enable factory data support on your device you still need to set the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` to ``y``.
```
```suggestion
      To enable factory data support on your device, you still need to set the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` to ``y``.
```

wiba-nordic commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -71,6 +71,19 @@ Peripheral samples
+
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD` kconfig option is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_FACTORY_DATA` instead to enable or disable creating the factory data set during building a Matter sample.
+      To enable factory data support on your device you still need to set the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` to ``y``.
+    * Factory data output files are located now in the ``<application_name>/zephyr/`` within the sample build directory.
```
```suggestion
    * Factory data output files are located now in the ``<application_name>/zephyr/`` directory within the sample build directory.
```

wiba-nordic commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -71,6 +71,19 @@ Peripheral samples
+
+  * With the inheritance of Zephyr's :ref:`zephyr:sysbuild` in the |NCS|, some changes are provided to the Matter samples and applications:
+
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD` kconfig option is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_FACTORY_DATA` instead to enable or disable creating the factory data set during building a Matter sample.
```
```suggestion
    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD` kconfig option is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_FACTORY_DATA` Kconfig option instead to enable or disable creating the factory data set during building a Matter sample.
```

wiba-nordic commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -71,6 +71,19 @@ Peripheral samples
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD` kconfig option is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_FACTORY_DATA` instead to enable or disable creating the factory data set during building a Matter sample.
+      To enable factory data support on your device you still need to set the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` to ``y``.
+    * Factory data output files are located now in the ``<application_name>/zephyr/`` within the sample build directory.
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_MERGE_WITH_FIRMWARE` kconfig option is deprecated and you need to use the :kconfig:option:`MATTER_FACTORY_DATA_MERGE_WITH_FIRMWARE` instead to enable or disable merging the factory data hex file with the final firmware hex file.
```
```suggestion
    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_MERGE_WITH_FIRMWARE` kconfig option is deprecated and you need to use the :kconfig:option:`MATTER_FACTORY_DATA_MERGE_WITH_FIRMWARE` Kconfig option instead to enable or disable merging the factory data HEX file with the final firmware HEX file.
```

wiba-nordic commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -71,6 +71,19 @@ Peripheral samples
+      To enable factory data support on your device you still need to set the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` to ``y``.
+    * Factory data output files are located now in the ``<application_name>/zephyr/`` within the sample build directory.
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_MERGE_WITH_FIRMWARE` kconfig option is deprecated and you need to use the :kconfig:option:`MATTER_FACTORY_DATA_MERGE_WITH_FIRMWARE` instead to enable or disable merging the factory data hex file with the final firmware hex file.
+    * :kconfig:option:`SB_CONFIG_MATTER_OTA` has been added to enable or disable generating Matter OTA package during the building process.
```
```suggestion
    * :kconfig:option:`SB_CONFIG_MATTER_OTA` Kconfig option has been added to enable or disable generating Matter OTA package during the building process.
```

wiba-nordic commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -71,6 +71,19 @@ Peripheral samples
+    * Factory data output files are located now in the ``<application_name>/zephyr/`` within the sample build directory.
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_MERGE_WITH_FIRMWARE` kconfig option is deprecated and you need to use the :kconfig:option:`MATTER_FACTORY_DATA_MERGE_WITH_FIRMWARE` instead to enable or disable merging the factory data hex file with the final firmware hex file.
+    * :kconfig:option:`SB_CONFIG_MATTER_OTA` has been added to enable or disable generating Matter OTA package during the building process.
+    * :kconfig:option:`CONFIG_CHIP_OTA_IMAGE_FILE_NAME` is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_OTA_IMAGE_FILE_NAME` kconfig option instead to define Matter OTA output filename.
```
```suggestion
    * :kconfig:option:`CONFIG_CHIP_OTA_IMAGE_FILE_NAME` Kconfig option is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_OTA_IMAGE_FILE_NAME` kconfig option instead to define Matter OTA output filename.
```

wiba-nordic commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -137,6 +151,10 @@ Matter
         1. Copy the content of the image configuration file :file:`prj.conf` located in the `sysbuild/<image_name>` directory (for example,  :file:`sysbuild/mcuboot`) to the :file:`prj.conf` file located in the :file:`child_image/<image_name>` directory.
         #. Copy the content of the board configuration file located in the :file:`sysbuild/<image_name>/boards` directory (for example, :file:`sysbuild/mcuboot/boards/nrf52840dk_nrf52840.conf`) to the board file located in the :file:`child_image/<image_name>/boards` directory.
 
+      * All Partition Manager configuration files (:file:`pm_static` files) with the suffix ``release`` has been removed from all samples.
```
```suggestion
      * All Partition Manager configuration files (:file:`pm_static` files) with the suffix ``release`` have been removed from all samples.
```

wiba-nordic commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -137,6 +151,10 @@ Matter
         #. Copy the content of the board configuration file located in the :file:`sysbuild/<image_name>/boards` directory (for example, :file:`sysbuild/mcuboot/boards/nrf52840dk_nrf52840.conf`) to the board file located in the :file:`child_image/<image_name>/boards` directory.
 
+      * All Partition Manager configuration files (:file:`pm_static` files) with the suffix ``release`` has been removed from all samples.
+        Those files are redundant now, since the new build system allows to use the file without the additional suffix if you use :makevar:`FILE_SUFFIX` if it is available in the projects directory.
```
```suggestion
        Those files are now redundant, since the new build system allows using the file without the additional suffix if you use :makevar:`FILE_SUFFIX` and it is available in the project's directory.
```

wiba-nordic commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -137,6 +151,10 @@ Matter
 
+      * All Partition Manager configuration files (:file:`pm_static` files) with the suffix ``release`` has been removed from all samples.
+        Those files are redundant now, since the new build system allows to use the file without the additional suffix if you use :makevar:`FILE_SUFFIX` if it is available in the projects directory.
+        It means that if you add ``-DFILE_SUFFIX=release`` to the cmake arguments while building the |NCS| Matter sample for example on ``nrf52840dk/nrf52840`` target the file ``pm_static_nrf52840dk_nrf52840.yaml`` will be used so the exact the same file ``pm_static_nrf52840dk_nrf52840_release.yaml`` is not needed anymore.
```
```suggestion
        For example, if you add ``-DFILE_SUFFIX=release`` to the CMake arguments while building an |NCS| Matter sample on the ``nrf52840dk/nrf52840`` target, the file :file:`pm_static_nrf52840dk_nrf52840.yaml` will be used.
        This means that the file :file:`pm_static_nrf52840dk_nrf52840_release.yaml` with the exact same contents is not needed anymore.
```

wiba-nordic commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -137,6 +151,10 @@ Matter
+      * All Partition Manager configuration files (:file:`pm_static` files) with the suffix ``release`` has been removed from all samples.
+        Those files are redundant now, since the new build system allows to use the file without the additional suffix if you use :makevar:`FILE_SUFFIX` if it is available in the projects directory.
+        It means that if you add ``-DFILE_SUFFIX=release`` to the cmake arguments while building the |NCS| Matter sample for example on ``nrf52840dk/nrf52840`` target the file ``pm_static_nrf52840dk_nrf52840.yaml`` will be used so the exact the same file ``pm_static_nrf52840dk_nrf52840_release.yaml`` is not needed anymore.
+        The :makevar:`CONF_FILE` is deprecated, but if you want to keep using it within your project you need to copy the content of the ``pm_static_nrf52840dk_nrf52840.yaml`` file, create the ``pm_static_nrf52840dk_nrf52840_release.yaml`` and paste it there.
```
```suggestion
        The :makevar:`CONF_FILE` argument is deprecated, but if you want to keep using it within your project, you need to create the :file:`pm_static_nrf52840dk_nrf52840_release.yaml` file and copy the content of the :file:`pm_static_nrf52840dk_nrf52840.yaml` file to it.
```

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-15686
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

kkasperczyk-no commented on applications/matter_weather_station/Kconfig.sysbuild:
```diff
@@ -4,18 +4,24 @@
+config NRF_DEFAULT_BLUETOOTH
+	default y if WIFI_NRF700X
+
+config NETCORE_IPC_RADIO_BT_HCI_IPC
```
Shouldn't it depend on the board selection?

kkasperczyk-no replied:
I mean that BT_HCI_IPC should be enabled only for the Thingy+nRF7002EB case, right? So why it is enabled always?

kkasperczyk-no commented on samples/matter/light_switch/Kconfig.sysbuild:
```diff
@@ -4,34 +4,34 @@
-
-config MATTER_FACTORY_DATA
-	default y if !BOARD_NRF21540DK
+	default y if SOC_SERIES_NRF53X
```
Ditto, do we want to enable BT_HCI_IPC for nRF5340DK and IEEE802154 for nRF7002DK?

ArekBalysNordic replied:
Right, will repair that

kkasperczyk-no commented on samples/matter/lock/README.rst:
```diff
@@ -126,15 +126,15 @@ You can program a portion of the application code related to the nRF70 Series' W
    .. code-block:: console
 
-      west build -b nrf5340dk/nrf5340/cpuapp -p -- -DSHIELD=nrf7002ek -Dmultiprotocol_rpmsg_SHIELD=nrf7002ek_coex -DFILE_SUFFIX=thread_wifi_switched -DCONFIG_NRF_WIFI_PATCHES_EXT_FLASH_STORE=y -Dmcuboot_CONFIG_UPDATEABLE_IMAGE_NUMBER=3
+      west build -b nrf5340dk/nrf5340/cpuapp -p -- -Dlock_SHIELD=nrf7002ek -Dmultiprotocol_rpmsg_SHIELD=nrf7002ek_coex -DFILE_SUFFIX=thread_wifi_switched -DSB_CONFIG_WIFI_PATCHES_EXT_FLASH_STORE=y -DSB_CONFIG_MCUBOOT_UPDATEABLE_IMAGES=3 -DCONFIG_CHIP_DFU_OVER_BT_SMP=y -DSB_CONFIG_WIFI_NRF700X=y -DSB_CONFIG_DFU_MULTI_IMAGE_PACKAGE_WIFI_FW_PATCH=y
```
It's a side question, so don't treat it as comment to your PR. I can see there are a lot of magic SB_ configs that we have to add in some cases, e.g. for the mcuboot, factory data or OTA, a now for enabling EK I think? What is the rule here? How the user is going to know that in some scenarios he has to add something to sysbuild and it others this is not required.

ArekBalysNordic replied:
Discussed f2f let's leave it as it is and if there are problems we will add more instructions there.

ArekBalysNordic commented on applications/matter_bridge/Kconfig.sysbuild:
```diff
@@ -4,29 +4,23 @@
 	default y
 
+# Enable IEEE802.15.4 serialization to network core
+config NETCORE_IPC_RADIO_IEEE802154
```
Because we need to have serialization to radio

ArekBalysNordic replied:
Oh, it is the bridge...

ArekBalysNordic replied:
Will remove it :)

nordicjm commented on applications/matter_bridge/Kconfig.sysbuild:
```diff
@@ -53,9 +47,16 @@ config DFU_MULTI_IMAGE_PACKAGE_APP
-config PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY
-	default y
+#### Support for WiFi patch on external flash memory.
+if BOARD_NRF5340DK_NRF5340_CPUAPP
+
+config MCUBOOT_UPDATEABLE_IMAGES
+	default 3
 
```
isn't this missing a wifi selection?

ArekBalysNordic replied:
In this sample, we only have WiFi support, so there is no need for a wifi selection anywhere within this file.

nordicjm commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -71,6 +71,19 @@ Peripheral samples
+  * With the inheritance of Zephyr's :ref:`zephyr:sysbuild` in the |NCS|, some changes are provided to the Matter samples and applications:
+
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD` kconfig option is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_FACTORY_DATA` Kconfig option instead to enable or disable creating the factory data set during building a Matter sample.
+      To enable factory data support on your device, you still need to set the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` to ``y``.
```
Really? A user should not have to set this, sysbuild should propagate that setting

ArekBalysNordic replied:
But one thing is to generate factory data, other to enable support for that in Matter sample. Sysbuild forced us to use this config but it is used only for generate factory data files. In fact sysbuild complicated our factory data.

nordicjm replied:
So should the SB option being set, force set the option set in the default image?

ArekBalysNordic replied:
OK, so the main thing is that we used to have two configs: CONFIG_CHIP_FACTORY_DATA that enables support for factory data and does not generate factory data set (users can do it using their scripts) and it was used just to enable parsing fd in the application. The second one: CONFIG_CHIP_FACTORY_DATA_BUILD was enabling the CMAKE script for generating factory data set. 
After moving to sysbuild only the second one became "deprecated" because you suggested creating the SB_CONFIG_MATTER_FACTORY_DATA config that only generates the factory data set. So now it is complicated, and we want to make it as easy as possible for customers and not provide many changes because they're using those mechanisms in their production. I thought that the easiest way is to inform users that CONFIG_CHIP_FACTORY_DATA_BUILD has been deprecated and now they must use SB_CONFIG_MATTER_FACTORY_DATA instead. But on the other hand, the SB_CONFIG_MATTER_FACTORY_DATA kconfig works only for sysbuild, so the old one must be available for non-sysbuild purposes, right? All other things are exactly the same, and they can manipulate the CONFIG_CHIP_FACTORY_DATA kconfig to enable/disable fd support. 

Are you seeing a better option?

ArekBalysNordic replied:
I will change this kconfig to `SB_CONFIG_MATTER_FACTORY_DATA_BUILD`, which describes the purpose better.

nordicjm replied:
does this need updating?

ArekBalysNordic replied:
Right, there is no _BUILD kconfig anymore.

ArekBalysNordic replied:
It is correct, Users need to use `SB_CONFIG_MATTER_FACTORY_DATA_GENERATE` to generate factory data, but apart from that they need to use `CONFIG_CHIP_FACTORY_DATA` to enable factory data support on SoC.

nordicjm commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -71,6 +71,19 @@ Peripheral samples
+
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD` kconfig option is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_FACTORY_DATA` Kconfig option instead to enable or disable creating the factory data set during building a Matter sample.
+      To enable factory data support on your device, you still need to set the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` to ``y``.
+    * Factory data output files are located now in the ``<application_name>/zephyr/`` directory within the sample build directory.
```
*are now located in

Would also not put sample build directory, just build directory, because it's the same for samples and user applications

nordicjm commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -71,6 +71,19 @@ Peripheral samples
+
+  * With the inheritance of Zephyr's :ref:`zephyr:sysbuild` in the |NCS|, some changes are provided to the Matter samples and applications:
+
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD` kconfig option is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_FACTORY_DATA` Kconfig option instead to enable or disable creating the factory data set during building a Matter sample.
```
deprecated is wrong, deprecated seems like it still works, but it does not

ArekBalysNordic replied:
Because it still works for non-sysbuild (child images) sample variant that as far as I know must be supported. But ok, deprecated may be misleading here.

ArekBalysNordic replied:
I re-thinked it and "deprecated" is correct here, because in fact it still must be used for the non-sysbuild variants. I will add a one-line comment there.

nordicjm commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -71,6 +71,19 @@ Peripheral samples
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD` kconfig option is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_FACTORY_DATA` Kconfig option instead to enable or disable creating the factory data set during building a Matter sample.
+      To enable factory data support on your device, you still need to set the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` to ``y``.
+    * Factory data output files are located now in the ``<application_name>/zephyr/`` directory within the sample build directory.
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_MERGE_WITH_FIRMWARE` kconfig option is deprecated and you need to use the :kconfig:option:`MATTER_FACTORY_DATA_MERGE_WITH_FIRMWARE` Kconfig option instead to enable or disable merging the factory data HEX file with the final firmware HEX file.
```
*Kconfig

nordicjm commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -137,6 +151,11 @@ Matter
 
+      * All Partition Manager configuration files (:file:`pm_static` files) with the suffix ``release`` have been removed from all samples.
+        Those files are now redundant, since the new build system allows using the file without the additional suffix if you use :makevar:`FILE_SUFFIX` and it is available in the project's directory.
+        For example, if you add ``-DFILE_SUFFIX=release`` to the CMake arguments while building an |NCS| Matter sample on the ``nrf52840dk/nrf52840`` target, the file :file:`pm_static_nrf52840dk_nrf52840.yaml` will be used.
```
*will be used as a fallback

nordicjm commented on samples/matter/lock/README.rst:
```diff
@@ -126,15 +126,15 @@ You can program a portion of the application code related to the nRF70 Series' W
 To prepare an application to use this feature, you need to create additional MCUboot partitions.
 To learn how to configure MCUboot partitions, see the :ref:`nrf70_fw_patch_update_adding_partitions` guide.
-To enable this feature for Matter, set the :kconfig:option:`CONFIG_NRF_WIFI_FW_PATCH_DFU` Kconfig option to ``y`` for the application (in the application :file:`prj.conf`) and the :kconfig:option:`CONFIG_UPDATEABLE_IMAGE_NUMBER` Kconfig option to ``3`` for the MCUBoot child image (in its own :file:`prj.conf`).
+To enable this feature for Matter, set the :kconfig:option:`SB_CONFIG_WIFI_PATCHES_EXT_FLASH_STORE`, and :kconfig:option:`SB_CONFIG_DFU_MULTI_IMAGE_PACKAGE_WIFI_FW_PATCH` Kconfig options to ``y`` for the application (in the application :file:`prj.conf`) and the :kconfig:option:`SB_CONFIG_MCUBOOT_UPDATEABLE_IMAGES` Kconfig option to ``3`` for the MCUBoot child image (in its own :file:`prj.conf`).
```
this goes in the sysbuild config, not the application config

kkasperczyk-no commented on applications/matter_weather_station/Kconfig.sysbuild:
```diff
@@ -4,18 +4,21 @@
+config NETCORE_IPC_RADIO_BT_HCI_IPC
+	default y if WIFI_NRF700X
+
+config NETCORE_IPC_RADIO_IEEE802154
```
I thought you will fix this btw of the above change. I think here we should go with `default y if !WIFI_NRF700X`

ArekBalysNordic replied:
Right, I didn't notice that, will fix it.

kkasperczyk-no commented on samples/matter/light_switch/Kconfig.sysbuild:
```diff
@@ -4,31 +4,37 @@
-
-config MATTER_FACTORY_DATA
-	default y if !BOARD_NRF21540DK
+	default y if SOC_SERIES_NRF53X
```
```suggestion
	default y if SOC_SERIES_NRF53X && WIFI_NRF700X
```
Shouldn't it be this way?

kkasperczyk-no commented on samples/matter/lock/Kconfig.sysbuild:
```diff
@@ -4,31 +4,37 @@
-
-config MATTER_FACTORY_DATA
-	default y if !BOARD_NRF21540DK
+	default y if (SOC_SERIES_NRF53X || SOC_SERIES_NRF54HX)
```
Ditto?
```suggestion
	default y if (SOC_SERIES_NRF53X || SOC_SERIES_NRF54HX) && WIFI_NRF700X
```

wiba-nordic commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -71,6 +71,23 @@ Peripheral samples
+
+  .. note::
+    
+    If you want to build sample without using sysbuild you need to use the old Kconfig options.
```
```suggestion
    If you want to build a sample without using sysbuild, you need to use the old Kconfig options.
```

nordicjm replied:
Not really sure why it's being documented how to build using child/parent at all, it is deprecated, it should not be used, it's included **ONLY** for customers to port their applications to sysbuild. If people want to use child/parent samples then they should stay using ncs 2.6

kkasperczyk-no replied:
Most likely, because we suspect that moving to sysbuild will be such a huge hit in the face for our customers that they would like to stay with 2.6.0, but they will have to use 2.7.0 to benefit from new Matter 1.3 revision. TBH majority of developers and QA engineers currently do not know how to move around in our SDK using new approach, so I would not expect it will be easy for our customers.

ArekBalysNordic replied:
It must not be deprecated. @LuDuda please comment on that.

nordicjm replied:
```
CMake Deprecation Warning at /tmp/bb/nrf/cmake/partition_manager.cmake:32 (message):
  

            ---------------------------------------------------------------------
            --- WARNING: Child and parent image functionality is deprecated   ---
            --- and should be replaced with sysbuild. Child and parent image  ---
            --- support remains only to allow existing customer applications  ---
            --- to build and allow porting to sysbuild, it is no longer       ---
            --- receiving updates or new features and it will not be possible ---
            --- to build using child/parent image at all in nRF Connect SDK   ---
            --- version 2.9 onwards.                                          ---
            ---------------------------------------------------------------------
```

LuDuda replied:
@nordicjm's assumption that customers do not migrate between revisions and must stay on NCS 2.6.0 is new to me, and it certainly does not reflect reality where our customers do migrate (e.g., for new Matter/Sidewalk/BLE versions, etc.). We see it also in customer surveys.

In NCS 2.7.0 and NCS 2.8.0, we need to enable our customers to use child images for their legacy out-of-tree applications. We have outlined a Matter-specific migration guide here, as we have eliminated the need to set the KConfig ROOT as previously requested. Therefore, this will require a manual migration by the customer.

Regarding the migration guide to sysbuild - we still need documentation here - so customers know how to do it. Ideally 
 I would love to have it even before the migration begins at Nordic. :)

Btw. Deprecation means that the feature will be soon removed and customers should not use it for new projects. But can keep it until is removed (3 releases by Nordic policy).

wbober replied:
Building using child-parent images is deprecated. Any references to this method should be clearly marked as such. The remark should point users to the sysbuild migration guide. Existing documentation referring to this build method should be kept until the deprecation period is over and support for this method is removed from the build system.

NordicBuilder commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| matter | https://github.com/nrfconnect/sdk-connectedhomeip/commit/579d8eebc3352ec1cc342ade2570897daa5ccdab | https://github.com/nrfconnect/sdk-connectedhomeip/pull/453 | [nrfconnect/sdk-connectedhomeip#453/files](https://github.com/nrfconnect/sdk-connectedhomeip/pull/453/files) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

ArekBalysNordic commented:
@nordicjm I changed the approach of our configs. We cannot use the new configs for Matter that were created for sysbuild because it breaks compatibility for our customers and we cannot allow doing that. Instead, I used the previous ones like CONFIG_CHIP_OTA_REQUESTOR and CONFIG_CHIP_FACTORY_DATA_BUILD to generate files in sysbuild's CMakeLists.txt.

nordicjm commented:
> @nordicjm I changed the approach of our configs. We cannot use the new configs for Matter that were created for sysbuild because it breaks compatibility for our customers and we cannot allow doing that. Instead, I used the previous ones like CONFIG_CHIP_OTA_REQUESTOR and CONFIG_CHIP_FACTORY_DATA_BUILD to generate files in sysbuild's CMakeLists.txt.

And by making that change you are:

1. Introducing a bug that I specifically avoided because sysbuild does not re-read the configuration of the application, so that means if a user configures an application with that option enabled or disabled, then changes it using menuconfig, the new setting will be completely ignored and the old one will be used instead
2. Putting a burden on my time for getting migration documentation complete

kkasperczyk-no commented:
> I never noticed but that help text uses the mcumgr go program

Do you mean our DFU related guide?

nordicjm commented:
> > I never noticed but that help text uses the mcumgr go program
> 
> Do you mean our DFU related guide on sdk-connectedhomeip?

https://github.com/nrfconnect/sdk-connectedhomeip/pull/453/files is where I saw it, buy any mentions of it. That tool should not be used by anyone

ArekBalysNordic commented:
> > @nordicjm I changed the approach of our configs. We cannot use the new configs for Matter that were created for sysbuild because it breaks compatibility for our customers and we cannot allow doing that. Instead, I used the previous ones like CONFIG_CHIP_OTA_REQUESTOR and CONFIG_CHIP_FACTORY_DATA_BUILD to generate files in sysbuild's CMakeLists.txt.
> 
> And by making that change you are:
> 
> 1. Introducing a bug that I specifically avoided because sysbuild does not re-read the configuration of the application, so that means if a user configures an application with that option enabled or disabled, then changes it using menuconfig, the new setting will be completely ignored and the old one will be used instead
> 2. Putting a burden on my time for getting migration documentation complete
> 3. I never noticed but that help text uses the mcumgr go program, that text needs to be completely removed and replaced without any mention of the junk tool

So it is a bug in sysbuild then because we are forced to completely new configs that conflict with the existing configuration.

kkasperczyk-no commented:
> > > I never noticed but that help text uses the mcumgr go program
> > 
> > 
> > Do you mean our DFU related guide on sdk-connectedhomeip?
> 
> https://github.com/nrfconnect/sdk-connectedhomeip/pull/453/files is where I saw it, but any mentions of it. That tool should not be used by anyone

> that text needs to be completely removed and replaced without any mention of the junk tool

Ok and who decides about it? I am not aware of any tool that would be officially recommended by our company to handle DFU on PC and this tool is mentioned in the Zephyr SMP server documentation: https://docs.zephyrproject.org/latest/samples/subsys/mgmt/mcumgr/smp_svr/README.html. Moreover it is used in CI tests that provides us with sufficient coverage. I agree that the tool is not perfect, but I'm afraid that the fact that you or I don't like this tool is not enough for us to remove it from guide.

doublemis1 commented:
> > > I never noticed but that help text uses the mcumgr go program
> > 
> > 
> > Do you mean our DFU related guide on sdk-connectedhomeip?
> 
> https://github.com/nrfconnect/sdk-connectedhomeip/pull/453/files is where I saw it, but any mentions of it. That tool should not be used by anyone

Is there any other cmd tool use for such purpose?
mcumgr tool is an official tool provided by the zephyr, I don;t see any deprecation on the tool and any other alternatives,
Verification by mobile phone is not possible in regression testing. Such sanity test (with mobile app) is executed by each release right before just to validate the procedure with mobile app. In our case DFU is crucial feature which need to be verified on regression testing (automatic).

wbober commented:
> Ok and who decides about it?

I do. 

The mcumgr is the command line tool supported by Nordic for the purpose of DFU updates for mcuboot.

nordicjm commented:
> > > @nordicjm I changed the approach of our configs. We cannot use the new configs for Matter that were created for sysbuild because it breaks compatibility for our customers and we cannot allow doing that. Instead, I used the previous ones like CONFIG_CHIP_OTA_REQUESTOR and CONFIG_CHIP_FACTORY_DATA_BUILD to generate files in sysbuild's CMakeLists.txt.
> > 
> > 
> > And by making that change you are:
> > 
> > 1. Introducing a bug that I specifically avoided because sysbuild does not re-read the configuration of the application, so that means if a user configures an application with that option enabled or disabled, then changes it using menuconfig, the new setting will be completely ignored and the old one will be used instead
> > 2. Putting a burden on my time for getting migration documentation complete
> > 3. I never noticed but that help text uses the mcumgr go program, that text needs to be completely removed and replaced without any mention of the junk tool
> 
> So it is a bug in sysbuild then because we are forced to completely new configs that conflict with the existing configuration.

It's not a bug in sysbuild, that's how sysbuild works. Otherwise you start entering the realm of child/parent image where you get continual reconfigure cycles. If something is local to an application then sysbuild should not reconfigure itself

nordicjm commented:
> @kkasperczyk-no
> 
> > Ok and who decides about it?
> 
> I do.
> 
> The mcumgr is the command line tool recommended by Nordic for the purpose of DFU updates for mcuboot.

You're recommended a dead, faulty tool which has the following notice

> This tool is provided for evaluation use only and is not recommended for use in a production environment. It has known issues and will not respect the MCUmgr protocol properly e.g. when an error is received, instead of aborting will, in some circumstances, sit in an endless loop of sending the same command over and over again. A universal replacement for this tool is currently in development and once released, support for the go tool will be dropped entirely. It is recommended that usage of tools listed above in the [Tools/libraries](https://docs.zephyrproject.org/latest/services/device_mgmt/mcumgr.html#mcumgr-tools-libraries) section are used instead of the go client.

wbober commented:
> > @kkasperczyk-no
> > > Ok and who decides about it?
> > 
> > 
> > I do.
> > The mcumgr is the command line tool recommended by Nordic for the purpose of DFU updates for mcuboot.
> 
> You're recommended a dead, faulty tool which has the following notice
> 
> > This tool is provided for evaluation use only and is not recommended for use in a production environment. It has known issues and will not respect the MCUmgr protocol properly e.g. when an error is received, instead of aborting will, in some circumstances, sit in an endless loop of sending the same command over and over again. A universal replacement for this tool is currently in development and once released, support for the go tool will be dropped entirely. It is recommended that usage of tools listed above in the [Tools/libraries](https://docs.zephyrproject.org/latest/services/device_mgmt/mcumgr.html#mcumgr-tools-libraries) section are used instead of the go client.

I'm well aware of the notice. Thank you.

The tool is used by a number of teams inside Nordic for development as well as in the CI infrastructure for end-to-end testing. This has been the case of many years. While it may have its shortcomings it serves its purpose well enough.

ArekBalysNordic commented:
@nordicjm @LuDuda @kkasperczyk-no  I reverted the changes and as we discussed we're going to leave SB_CONFIG_MATTER... configs for factory data and OTA, and I wrote the limitation in several places - in the migration guide, factory data guide and all sample README files. Please review it once again.

wiba-nordic commented on samples/matter/lock/README.rst:
```diff
@@ -291,10 +291,14 @@ This means that a new factory data set will be automatically generated when buil
 
+.. important::
+
+    If you modify any of the factory data-related Kconfig option you need to build a Matter sample with the pristine option to clear the CMake cache or clear the cache manually.
```
```suggestion
    If you modify any of the factory data-related Kconfig options, you need to either build the sample with the pristine option to clear the CMake cache or clear the cache manually.
```

wiba-nordic commented on applications/matter_weather_station/README.rst:
```diff
@@ -204,33 +204,36 @@ To build this application to work with the nRF7002 Wi-Fi expansion board:
 
+.. important::
+
+    If you modify any of the factory data-related Kconfig option you need to build a Matter sample with the pristine option to clear the CMake cache or clear the cache manually.
```
```suggestion
    If you modify any of the factory data-related Kconfig options, you need to either build the application with the pristine option to clear the CMake cache or clear the cache manually.
```

wiba-nordic commented on doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:
```diff
@@ -71,6 +71,24 @@ Peripheral samples
+    * :kconfig:option:`SB_CONFIG_MATTER_OTA` Kconfig option has been added to enable or disable generating Matter OTA package during the building process.
+    * :kconfig:option:`CONFIG_CHIP_OTA_IMAGE_FILE_NAME` Kconfig option is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_OTA_IMAGE_FILE_NAME` Kconfig option instead to define Matter OTA output filename.
+
+    If you modify any of the factory data-related Kconfig option you need to build a Matter sample with the pristine option to clear the CMake cache.
```
```suggestion
    If you modify any of the factory data-related Kconfig options, you need to either build the sample or application with the pristine option to clear the CMake cache or clear the cache manually.
```
Feel free to leave out the manual clearing option if it was left out of the migration guide on purpose.

ArekBalysNordic replied:
I will leave it because somebody can for example remove the build directory manually and in this way, remove the cache.

nordicjm replied:
This can actually go entirely, I tested and if these options are changed, they are picked up and a new image is generated. Must have been an issue that has been fixed sometime without being aware

ArekBalysNordic replied:
Ok, so I will remove the note from all places.

ArekBalysNordic replied:
I've removed all those lines that inform about the pristine build.

LuDuda commented:
FYI. Have to be merged after: https://github.com/nrfconnect/sdk-nrf/pull/15836

ArekBalysNordic commented:
@nordicjm can we merge https://github.com/nrfconnect/sdk-connectedhomeip/pull/453 and update SHA?
